openapi/workflow_execution_service.openapi.yaml

```diff
@@ -639,7 +644,33 @@ components:
           type: string
         state:
           $ref: '#/components/schemas/State'
-      description: Small description of a workflow run, returned by server during listing
+      description: State information of a workflow run
+    RunSummary:
+      title: RunSummary
+      allOf:
+        - $ref: '#/components/schemas/RunStatus'
+        - type: object
+          properties:
+            name:
+              type: string
+              description: The workflow name
+            start_time:
+              type: string
+              description: When the run started executing, in ISO 8601 format "%Y-%m-%dT%H:%M:%SZ"
+            end_time:
+              type: string
+              description: When the run stopped executing (completed, failed, or cancelled), in ISO 8601 format "%Y-%m-%dT%H:%M:%SZ"
+            tags:
+              type: object
+              description: Arbitrary key/value tags added by the client during run creation 
+              additionalProperties:
+                type: string
+          required:
+            - name
+            - start_time
+            - end_time
+            - tags
```

Comment:
You are probably right.

But then we could set defaults for these values (e.g., empty strings). That way we would perhaps encourage implementers a little more to give their best at producing these values, wherever possible. And on the client side we could rely on these fields being available.

Reply:
In general I prefer `null` to an empty value like `""` (empty maps are fine).  Empty strings end up requiring a bunch of edge case handling on the client side, ie in typed systems you now need to ensure that non null values can actually be deserialized as the expected type.

For example, I can represent a `RunSummary` in java like the following: 
```java
public record RunSummary(String id, String name, Instant start_time, Instant end_time, Map<String,String> tags){
}
```

Now, if I actually wanted to use this record when receiving responses, I would not be able to directly deserialize the value directly, but would now need to have a "holding" class because `""` is not a valid Date. Trying to deserialize the following using a library like `Jackson` would result in a `JsonProcessingException`, OR a `DateTimeParseException`. 
```
{
  "id": "foo",
  "name": "bar",
  "start_time": "",
  "end_time": ""
  "tags": {}
} 
```

Whereas the following would work:
```
{
  "id": "foo",
  "name": "bar",
  "start_time": null,
  "end_time": null
  "tags": {}
} 
```

Reply:
I don't think `name` requires parsing of a workflow. As I mentioned on a separate comment, `name` is effectively a property of `RunRequest` (possibly defined as a `tag`) - that is it is the name of the **run** and not the name of the **workflow**. For that I think I'll backtrack a bit and say that `tags` is the better place to hold this information and that **both** a run name and workflow name are a good idea.

For example:
- workflow names can be:
    - nf-core-sarek-3.1.2
    - gatk4-variant-discovery
- run names can be:
    - sample-id-12345
    - lims-id-{{uuid}}

Having these as tags presents a way to filter the runs list response along the lines of:
- "how many runs of a specific workflow are there?"
- "how many workflows or runs are processing data for a specific sample?"

Reply:
We would already have unique identifiers for runs (the `run_id`). Having an optional human-readable alternative to these is useful, I think, but would indeed go well in `tags`, as it's really just an alternative.

Workflow names, on the other hand, are extremely useful. I would venture to say that these would be among the features which users filter there reads by most often, if not the most commonly used of all. If I run a lot of analyses, next to inputs, I think that the workflow "name" is really what I would look for first to narrow things down. And with that, I think we could indeed add an optional `workflow_name` to `RunRequest` (I think TES has something similar) - and then encourage implementers to provide a reasonable default (e.g., repo name, TRS resource name, workflow URL suffix) if a workflow name is not provided by the client. For this, `tags` feels a little too optional for me personally.

However, wouldn't strongly object to having both in `tags`.

Reply:
If `name` is a property of `RunSummary` I'd prefer the spect to be clear about what item is being named. If part of `RunSummary`, my expectation is that it is the name of the run and not the name of the workflow the run invoked.

With just workflow name, and assuming `tags` doesn't provide additional human readable information to help disambiguate, a list response would look like:
```json
[
    {
        "id": "e81da9ff-f5a8-4f20-9d5f-772efc4a93b4",
        "start_time": "YYYY-MM-DD HH:MM:SS Z",
        "end_time": "YYYY-MM-DD HH:MM:SS Z",
        "name": "nf-core-rnaseq",
        "tags": { ... }
    },
    {
        "id": "1ae61546-7cfa-4959-bf28-247ce1390508",
        "start_time": "YYYY-MM-DD HH:MM:SS Z",
        "end_time": "YYYY-MM-DD HH:MM:SS Z",
        "name": "nf-core-rnaseq",
        "tags": { ... }
    },
    {
        "id": "c9be874b-5b9c-48b3-879e-40986fe9dbf0",
        "start_time": "YYYY-MM-DD HH:MM:SS Z",
        "end_time": "YYYY-MM-DD HH:MM:SS Z",
        "name": "nf-core-rnaseq",
        "tags": { ... }
    },
    {
        "id": "86aaf850-b2c4-4c9d-b898-16ed3ce1daba",
        "start_time": "YYYY-MM-DD HH:MM:SS Z",
        "end_time": "YYYY-MM-DD HH:MM:SS Z",
        "name": "nf-core-rnaseq",
        "tags": { ... }
    }
]
```

This would be challenging for an end-user to find the specific run that say processed sample "LIMSJVLO6YWS". Omitting `name` from the top level of `RunSummary` and having tags explicitly for `workflow_name` and `run_name` looks like:

```json
[
    {
        "id": "e81da9ff-f5a8-4f20-9d5f-772efc4a93b4",
        "start_time": "YYYY-MM-DD HH:MM:SS Z",
        "end_time": "YYYY-MM-DD HH:MM:SS Z",
        "tags": {
            "workflow_name": "nf-core-rnaseq",
            "run_name": "LIMS2ZR5ZS29"
        }
    },
    {
        "id": "1ae61546-7cfa-4959-bf28-247ce1390508",
        "start_time": "YYYY-MM-DD HH:MM:SS Z",
        "end_time": "YYYY-MM-DD HH:MM:SS Z",
        "tags": {
            "workflow_name": "nf-core-rnaseq",
            "run_name": "LIMSXXOFXDMW"
        }
    },
    {
        "id": "c9be874b-5b9c-48b3-879e-40986fe9dbf0",
        "start_time": "YYYY-MM-DD HH:MM:SS Z",
        "end_time": "YYYY-MM-DD HH:MM:SS Z",
        "tags": {
            "workflow_name": "nf-core-rnaseq",
            "run_name": "LIMSJVLO6YWS"
        }
    },
    {
        "id": "86aaf850-b2c4-4c9d-b898-16ed3ce1daba",
        "start_time": "YYYY-MM-DD HH:MM:SS Z",
        "end_time": "YYYY-MM-DD HH:MM:SS Z",
        "tags": {
            "workflow_name": "nf-core-rnaseq",
            "run_name": "LIMS42IXEIFO"
        }
    }
]
```

IMO this provides a more useful response and maintains flexibility for clients to display or filter information how they need.

A compromise would be to maintain semantic consistency where the `name` property in RunSummary is the `run_name` and `workflow_name` is additional metadata provided in `tags`:

```json
[
    {
        "id": "e81da9ff-f5a8-4f20-9d5f-772efc4a93b4",
        "start_time": "YYYY-MM-DD HH:MM:SS Z",
        "end_time": "YYYY-MM-DD HH:MM:SS Z",
        "name": "LIMS2ZR5ZS29",
        "tags": {
            "workflow_name": "nf-core-rnaseq"
        }
    },
    {
        "id": "1ae61546-7cfa-4959-bf28-247ce1390508",
        "start_time": "YYYY-MM-DD HH:MM:SS Z",
        "end_time": "YYYY-MM-DD HH:MM:SS Z",
        "name": "LIMSXXOFXDMW",
        "tags": {
            "workflow_name": "nf-core-rnaseq"
        }
    },
    {
        "id": "c9be874b-5b9c-48b3-879e-40986fe9dbf0",
        "start_time": "YYYY-MM-DD HH:MM:SS Z",
        "end_time": "YYYY-MM-DD HH:MM:SS Z",
        "name": "LIMSJVLO6YWS",
        "tags": {
            "workflow_name": "nf-core-rnaseq"
        }
    },
    {
        "id": "86aaf850-b2c4-4c9d-b898-16ed3ce1daba",
        "start_time": "YYYY-MM-DD HH:MM:SS Z",
        "end_time": "YYYY-MM-DD HH:MM:SS Z",
        "name": "LIMS42IXEIFO",
        "tags": {
            "workflow_name": "nf-core-rnaseq"
        }
    }
]

```